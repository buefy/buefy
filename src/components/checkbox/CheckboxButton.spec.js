import { shallowMount } from '@vue/test-utils'
import BCheckboxButton from '@components/checkbox/CheckboxButton'

let wrapper

describe('BCheckboxButton', () => {
    beforeEach(() => {
        wrapper = shallowMount(BCheckboxButton)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('BCheckboxButton')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('has an input checkbox', () => {
        expect(wrapper.contains('label input[type=checkbox]')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        const NEW_VALUE = true

        wrapper.setProps({ value: NEW_VALUE })

        expect(wrapper.vm.newValue).toBeTruthy()
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(NEW_VALUE)
    })
})
