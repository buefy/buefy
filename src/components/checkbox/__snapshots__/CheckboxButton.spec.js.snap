// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCheckboxButton render correctly 1`] = `<div class="control"><label class="b-checkbox checkbox button is-primary"> <input type="checkbox" value="false"></label></div>`;
